py3dtiles/convert.py

```diff
@@ -290,35 +292,93 @@ def init_parser(subparser, str2bool):
+                   jobs=args.jobs,
+                   cache_size=args.cache_size,
+                   srs_out=args.srs_out,
+                   srs_in=args.srs_in,
+                   fraction=args.fraction,
+                   benchmark=args.benchmark,
+                   rgb=args.rgb,
+                   graph=args.graph,
+                   color_scale=args.color_scale,
+                   verbose=args.verbose)
+
+
+def convert(files,
+            outfolder='./3dtiles',
+            overwrite=False,
+            jobs=multiprocessing.cpu_count(),
+            cache_size=int(total_memory_MB / 10),
+            srs_out=None,
+            srs_in=None,
+            fraction=100,
+            benchmark=None,
+            rgb=True,
+            graph=False,
+            color_scale=None,
+            verbose=False):
+    """convert
+
+    Convert pointclouds (xyz or las) to 3dtiles tileset containing pnts node
+
+    :param files: Filenames to process. The file must use the .las or .xyz format.
+    :type files: list of str
+    :param outfolder: The folder where the resulting tileset will be written.
+    :type outfolder: path-like object
+    :param overwrite: Overwrite the ouput folder if it already exists.
+    :type overwrite: bool
+    :param jobs: The number of parallel jobs to start. Default to the number of cpu.
+    :type jobs: int
+    :param cache_size: Cache size in MB. Default to available memory / 10.
+    :type cache_size: int
+    :param srs_out: SRS to convert the output with (numeric part of the EPSG code)
+    :type srs_out: int or str
+    :param srs_in: Override input SRS (numeric part of the EPSG code)
+    :type srs_in: int or str
+    :param fraction: Percentage of the pointcloud to process, between 0 and 100.
+    :type fraction: int
+    :param benchmark: Print summary at the end of the process
+    :type benchmark: str
+    :param rgb: Export rgb attributes.
+    :type rgb: bool
+    :param graph: Produce debug graphes (requires pygal).
+    :type graph: bool
+    :param color_scale: Force color scale
+    :type color_scale: float
+
+    """
 
     # create folder
-    if os.path.isdir(folder):
-        if args.overwrite:
-            shutil.rmtree(folder)
+    if os.path.isdir(outfolder):
+        if overwrite:
+            shutil.rmtree(outfolder)
         else:
-            print('Error, folder \'{}\' already exists'.format(folder))
+            print('Error, folder \'{}\' already exists'.format(outfolder))
             sys.exit(1)
 
-    os.makedirs(folder)
-    working_dir = folder + '/tmp'
+    os.makedirs(outfolder)
+    working_dir = os.path.join(outfolder, '/tmp')
     os.makedirs(working_dir)
 
     node_store = SharedNodeStore(working_dir)
 
     # read all input files headers and determine the aabb/spacing
-    _, ext = os.path.splitext(args.files[0])
+    _, ext = os.path.splitext(files[0])
     fn = las_reader.init if ext == '.las' else xyz_reader.init
-    infos = fn(args)
```

Comment:
I suggest a more explicit function name, like `reader`, instead of `fn`.

Reply:
I chose `init_reader_fn`